lib/wor/requests/base.rb

```diff
@@ -0,0 +1,151 @@
+require 'uri'
+require 'httparty'
+require 'json'
+
+# rubocop:disable ClassLength
+module Wor
+  module Requests
+    class Base
+      VALID_HTTP_VERBS = [:get, :post, :patch, :put, :delete].freeze
+      # According to RFC 7231
+      VALID_COMMON_ATTRIBUTES = [:path, :headers, :attempting_to, :response_type].freeze
+      VALID_GET_ATTRIBUTES    = VALID_COMMON_ATTRIBUTES + [:query]
```

Comment:
Maybe we should need to freeze these arrays? I think that the concatenation of a frozen array with a non-frozen array returns a non-frozen array U_U